### PR TITLE
Extension maintanance

### PR DESCRIPTION
**Pull request by marcel1hruska**
- added npm tests (also to pipeline)
- refactored extension

### CODE REVIEW

File: .github/workflows/alpine.yml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
         run: mkdir build && cd build && cmake -G Ninja ../
       - name: Build
         run: cd build && cmake --build .
-      - name: Test
+      - name: Server Test
```
Comment (michalbali256) [inline .github/workflows/alpine.yml:40]: Is it possible on alpine to start at least those tests that do not require gui?
Or do all require gui?

File: clients/vscode-hlasmplugin/package.json [hunk 2]
```diff
@@ -187,7 +188,7 @@
       "type": "object",
       "title": "hlasmplugin configuration",
       "properties": {
-        "semanticHighlightingColors": {
+        "hlasmplugin.semanticHighlightingColors": {
```
Comment (michalbali256) [inline clients/vscode-hlasmplugin/package.json:191]: Are we still using name hlasmplugin? Maybe just hlasm or HlasmLanguageSupport

Comment (marcel1hruska) [reply clients/vscode-hlasmplugin/package.json:191]: all other properties have that name. but simply 'hlasm' seems more appropriate

File: .github/workflows/alpine.yml (hunk 1 above)
Comment (marcel1hruska) [reply .github/workflows/alpine.yml:40]: The whole testing suite requires gui. Thats the vscode way

Comment (michalbali256) [reply .github/workflows/alpine.yml:40]: OK. In the future, we should find a way to test it with theia & selenium

File: clients/vscode-hlasmplugin/.gitignore [hunk 3]
```diff
@@ -1,4 +1,4 @@
 node_modules
 lib
 hlasmplugin.log
-bin
```
Comment (michalbali256) [inline clients/vscode-hlasmplugin/.gitignore:4]: Why is bin removed? The contents of bin (language server executables) should never be checked into the git

Comment (marcel1hruska) [reply clients/vscode-hlasmplugin/.gitignore:4]: This change for the build/bin folder, not bin/. Im not really sure why but it works, the build/bin files were updated and bin/ was not uploaded.

**Pull request merged**